### PR TITLE
Propagator: Make sure we schedule only one job #7439

### PR DESCRIPTION
**Pull request by guruz**
To not starve the event loop.

(Idea by @ogoffart )

### CODE REVIEW

File: src/libsync/owncloudpropagator.cpp [hunk 1]
```diff
@@ -591,6 +592,8 @@ QString OwncloudPropagator::getFilePath(const QString &tmp_file_name) const
 
 void OwncloudPropagator::scheduleNextJob()
 {
+    if (m_jobScheduled>0) return; // don't schedule more than 1
```
Comment (ckamm) [inline src/libsync/owncloudpropagator.cpp]: minor: would prefer " > "

File: src/libsync/owncloudpropagator.h [hunk 2]
```diff
@@ -571,6 +572,7 @@ private slots:
     AccountPtr _account;
     QScopedPointer<PropagateDirectory> _rootJob;
     SyncOptions _syncOptions;
+    int m_jobScheduled;
```
Comment (ckamm) [inline src/libsync/owncloudpropagator.h]: minor: would prefer = 0; here instead of in ctor
minor: we use only `_`, not `m_` in OCC

**Comment by ckamm** [comment]
oh, maybe also set to 0 if it's ever negative, just for sanity

**Comment by ogoffart** [comment]
Looks ok, but it need to be tested.
It could simply be a `bool` 
As for the negative, it should be an assert, perhaps

**Pull request merged**